feat: facade and entry point calls for service catalog

arcrank opened the pull request:
Fixes #

dponzo commented on src/service-catalog.ts:
```diff
@@ -166,4 +166,44 @@ export async function describeProvisioningParameters(productId: string, provisio
+ * @param options
+ * @returns ProductVersionData aggregate of provisioning artifact details.
+ */
+export async function describeProvisionedProduct( options: DescribeProvisionedProductOptions = {}): Promise<ProductDataAggregate> {
```
this should be renamed, either describeProduct or Product version

ProvisionedProduct not really accurate

dponzo committed:
Add full describe call path and modify APIs

This commit does not include any reachable code paths.
We are opting to reduce number of needed API calls which we originally had before
with multiple calls to fetch provisioning artifacts and launch paths separately in order to
handle conerns over pagination. Instead we will just call describe product which returns
everything in the response body with minimal latency differences.

This also sets up adding both ability to describe a single product, as well as describing
all available products, which we have default paths for selecting the PA and launch path for.
